dds_web/config.py

```diff
@@ -42,6 +42,17 @@ class Config(object):
     # Devel settings
     TEMPLATES_AUTO_RELOAD = True
 
+    # Request parameters dropped from the logfile
+    SENSITIVE_REQUEST_ARGS = [
+        "checksum",
+        "compressed",
+        "public_key",
+        "size",
+        "size_processed",
+        "salt",
+        "token",
+    ]
```

Comment:
This assumes though that the args are always written in this exact way. So if (for some reason) the variable is actually in caps, then this won't catch that that information is actually sensitive and still log it. So in that case we would need to check caps as well, and possibly also misspelled args. Shouldn't happen of course, just thinking different scenarios and worried that something will accidentally be logged. Also, someone could potentially post a huge amount of json to the server, which does not include these args, and that will let everything be logged. This applies to all endpoints though so we should probably think about if we really need json to be logged and, if so, I'd suggest specifying what actually _should_ be logged instead of what shouldn't.

Reply:
The objections have been rendered irrelevant by the current decision to not log JSON and arbitrary arguments, but I would nevertheless like to respond briefly: 

_So in that case we would need to check caps as well, and possibly also misspelled args_
I fail to see how that is a problem. If we by mistake have an error in our CLI/API code, we should fix it to be consistent across our endpoints (and also would have to, because Marshmallow would also complain).

_and worried that something will accidentally be logged._ 
In log files that are only accessible to the Super Admins who have full database access. Admittedly, having useless log files that don't help to investigate potential issues are a larger concern to me than having a Super Admin knowing to much about their system.

_Also, someone could potentially post a huge amount of json to the server, which does not include these args, and that will let everything be logged_
It would have been possible to use `len()` to drop to large chunks of information. URLs however also have no limit and can be between >8k and sometimes up to <64k characters long (don't know about Flasks limit), so also logging the project arg could be misused to flood the log with junk information.

Reply:
> So in that case we would need to check caps as well, and possibly also misspelled args
I fail to see how that is a problem. If we by mistake have an error in our CLI/API code, we should fix it to be consistent across our endpoints (and also would have to, because Marshmallow would also complain).

True. Just thinking about potential issues.

> and worried that something will accidentally be logged.
In log files that are only accessible to the Super Admins who have full database access. Admittedly, having useless log files that don't help to investigate potential issues are a larger concern to me than having a Super Admin knowing to much about their system.

Yes, of course we need useful log files. But again, the action loggers purpose is to just see who or what has attempted what endpoint in what project. So having the possibility of logging json information just adds risk and defeats the purpose of this specific logger. The general logger, however, will log all information regarding potential errors where we can investigate potential issues when there are any. (@alneberg I believe I misspoke yesterday or Wednesday, so just to clarify: The logging occurs no matter what, even if the user is not authenticated. This is because we need to log when a request is denied and be able to figure out why, but also here the json is denied because the user has not been granted access to the system.)

> Also, someone could potentially post a huge amount of json to the server, which does not include these args, and that will let everything be logged
It would have been possible to use len() to drop to large chunks of information. URLs however also have no limit and can be between >8k and sometimes up to <64k characters long (don't know about Flasks limit), so also logging the project arg could be misused to flood the log with junk information.

Sure, that could happen. There's a lower risk of someone passing in a 64k character long project id though then someone passing in a lot of data in the request. But yes, we can add a limit to the length of the project ID being logged. 👍🏻  Lets think about it first though, don't do this now.

Anyway, remove the `SENSITIVE_REQUEST_ARGS`. 👍🏻

Reply:
The `SENSITIVE_REQUEST_ARGS` seem to still be there so just remove that